power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -176,7 +177,10 @@ inline auto build_transformer_graph(State const& state) -> TransformerGraph {
                                  edge_props.cbegin(),
                                  static_cast<TrafoGraphIdx>(state.components.template size<Node>())};
 
-    BGL_FORALL_VERTICES(v, trafo_graph, TransformerGraph) { trafo_graph[v].is_source = false; }
+    BGL_FORALL_VERTICES(v, trafo_graph, TransformerGraph) {
+        auto const out_degree = boost::out_degree(v, trafo_graph);
```

Comment:
@Jerry-Jinfeng-Guo What does out_degree do?

Reply:
It's debug line that I forgot to remove ;)